specs/bond-manager.md

```diff
@@ -0,0 +1,111 @@
+**Table of Contents**  *generated with [DocToc](https://github.com/thlorenz/doctoc)*
+
+- [Overview](#overview)
+- [The Bond Problem](#the-bond-problem)
+  - [Simple Bond](#simple-bond)
+  - [Variable Bond](#variable-bond)
+- [Bond Manager Interface](#bond-manager-interface)
+- [Bond Manager Implementation](#bond-manager-implementation)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Overview
+
+In the context of permissionless output proposals, bonds are a value that must
+be attached to an output proposal. In this case, the bond will be paid in ether.
+
+By requiring a bond to be posted with an output proposal, spam and invalid outputs
+are disincentivized. It's important to note that it is directly disincentivized
+because if the output is invalid, it will be seized by a set of challenge agents
+when the output is deleted. Thus, the bond acts as a form of economic security.
+
+Concretely, outputs will be permissionlessly proposed to the `L2OutputOracle` contract.
+When calling the propose function, the ether value is sent as the bond. This bond is
+then held by a bond manager contract. The bond manager contract is responsible for
+both the [dispute-games](./dispute-game.md) and the `L2OutputOracle` (further detailed
+in [proposals](./proposals.md)).
+
+The bond manager will need to handle bond logic for a variety of different
+[dispute-games](./dispute-game-interface.md). In the simplest "attestation" dispute game,
+bonds will not be required since the attestors are permissioned. But in more complex games,
+such as the fault dispute game, challengers and defenders perform a series of
+alternating onchain transactions requiring bonds at each step.
+
+## The Bond Problem
+
+At its core, the bond manager is straightforward - it escrows or holds ether and returns
+it at maturity or seized. But the uncertainty of introducing bonds lies in the
+bond _sizing_, i.e. how much should a bond be? Sizing bonds correctly is a function of
+the bond invariant: the bond must be greater than or equal to the cost of the next step.
+If bonds are priced too low, then the bond invariant is violated and there isn't an economic
+incentive to execute the next step. If bonds are priced too high, then the actors posting
+bonds can be priced out.
+
+Below, we outline two different approaches to sizing bonds and the tradeoffs of each.
+
+### Simple Bond
+
+The _Simple Bond_ is a very conservative approach to bond management, establishing a **fixed** bond
+size. The idea behind simple bond pricing is to establish the worst case gas cost for
+the next step in the dispute game.
+
+With this approach, the size of the bond is computed up-front when a dispute game is created.
+For example, in an attestation dispute game, this bond size can be computed using the following
+simple linear function:
+```
+size = number of signers * gas used to progress game + security overhead
+```
```

Comment:
real quick, thinking we may want to alter the order of operations

```
size = number of signers * (gas used to progress game + security overhead)
```

So there is a security overhead attached with each move in the game instead of a single value added as the overhead, this would mean the security overhead is equally as effective as the number of signers increases